internal/exportimport/scheduler.go

```diff
@@ -115,7 +115,21 @@ func syncFilesFromIndex(ctx context.Context, db *exportimportdb.ExportImportDB,
 			// drop blank lines.
 			continue
 		}
-		proposedURL := fmt.Sprintf("%s%s", config.ExportRoot, strings.TrimSpace(zipFile))
+		// Parse the export root to see if there is a defined path element.
+		base, err := url.Parse(config.ExportRoot)
```

Comment:
This is fine, but isn't this the logic that `path.Join` does? I'm confused why we need to implement this ourselves.

Reply:
*cough* unit test *cough*

(I hate being that guy...)

Reply:
Please continue to be that person 😄

Reply:
still need to parse, because we only want to path.Join on the path elements, and not protocol.

added unit test to cover error conditions - happy path was well covered.